Semester5/Formal Languages and Compiler Design/lab2/Lexic.txt

```diff
@@ -0,0 +1,39 @@
+Alphabet:
+
+	a. Upper (A-Z) and lower case letters (a-z) of the English alphabet
+	b. Underline character '_';
+	c. Decimal digits (0-9);
+
+Lexic:
+
+	a. Special symbols
+		-operators +, -, *, /, ==, <, >, <=, >=, =, !=
+		-separators [], {}, (), ;, space, "",
+		-reserved words: Integer, String, Char, for, return, in, out, if, elif, else, ArrayList, while
+	
+	b. Identifiers
+		Sequence of letters, digits or "_", with the condition the first character is a letter.
+		
+		<identifier> ::= <letter> | <identifier><letter> | <identifier>_<identifier> | <identifier><digit>
```

Comment:
<identifier>_<identifier> is not correct because the recursion won t work, it will be <identifier>_